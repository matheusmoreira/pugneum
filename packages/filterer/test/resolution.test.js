const path = require('path');
const filename = path.basename(__filename);

const lex = require('pugneum-lexer');
const parse = require('pugneum-parser');
const filter = require('../');

test('installed filter packages can be used implicitly', () => {

  const src = `
pre
  code
    :'highlight.js'(language=ruby)
      puts 'This should be', :syntax_highlighted
`;

  const tokens = lex(src, {filename});
  const ast = parse(tokens, {filename, src});
  const filtered = filter(ast);

  expect(filtered).toMatchSnapshot();
});
