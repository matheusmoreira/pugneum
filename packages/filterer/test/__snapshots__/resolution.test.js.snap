// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`installed filter packages can be used implicitly 1`] = `
{
  "filename": "resolution.test.js",
  "line": 0,
  "nodes": [
    {
      "attributeBlocks": [],
      "attrs": [],
      "block": {
        "filename": "resolution.test.js",
        "line": 2,
        "nodes": [
          {
            "attributeBlocks": [],
            "attrs": [],
            "block": {
              "filename": "resolution.test.js",
              "line": 3,
              "nodes": [
                {
                  "attrs": [
                    {
                      "column": 21,
                      "filename": "resolution.test.js",
                      "line": 4,
                      "mustEscape": true,
                      "name": "language",
                      "val": "ruby",
                    },
                  ],
                  "block": {
                    "filename": "resolution.test.js",
                    "line": 4,
                    "nodes": [
                      {
                        "column": 7,
                        "filename": "resolution.test.js",
                        "line": 5,
                        "type": "Text",
                        "val": "puts 'This should be', :syntax_highlighted",
                      },
                    ],
                    "type": "Block",
                  },
                  "column": 5,
                  "filename": "resolution.test.js",
                  "line": 4,
                  "name": "highlight.js",
                  "type": "Text",
                  "val": "puts <span class="hljs-string">&#x27;This should be&#x27;</span>, <span class="hljs-symbol">:syntax_highlighted</span>",
                },
              ],
              "type": "Block",
            },
            "column": 3,
            "filename": "resolution.test.js",
            "isInline": true,
            "line": 3,
            "name": "code",
            "selfClosing": false,
            "type": "Tag",
          },
        ],
        "type": "Block",
      },
      "column": 1,
      "filename": "resolution.test.js",
      "isInline": false,
      "line": 2,
      "name": "pre",
      "selfClosing": false,
      "type": "Tag",
    },
  ],
  "type": "Block",
}
`;
